test/gtest/uct/test_amo_fand_for.cc

```diff
@@ -0,0 +1,119 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2018.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "test_amo.h"
+
+
+class uct_amo_fand_for_test : public uct_amo_test {
+public:
+
+    template <typename T>
+    void test_fop(send_func_t send, T (*op)(T, T)) {
+        /*
+         * Method: Do concurrent atomic fetch-and-and/or of constant random value
+         * to a single atomic variable. Check that every sender gets a unique reply
+         * and the final value of atomic variable is the and/or of all.
+         */
+
+        mapped_buffer recvbuf(sizeof(T), 0, receiver());
+
+        T value = rand64();
+        T add   = rand64();
+        *(T*)recvbuf.ptr() = value;
+
+        std::vector<uint64_t> exp_replies;
+        for (unsigned i = 0; i < count() * num_senders(); ++i) {
+            exp_replies.push_back(value);
+            value = op(value, add);
+        }
+
+        run_workers(send, recvbuf, std::vector<uint64_t>(num_senders(), add), false);
+
+        validate_replies(exp_replies);
+
+        wait_for_remote();
+        EXPECT_EQ(value, *(T*)recvbuf.ptr());
+    }
+};
+
+template <typename T>
+T and_op(T v1, T v2)
+{
+    return v1 & v2;
+}
+
+template <typename T>
+T or_op(T v1, T v2)
+{
+    return v1 | v2;
+}
+
+UCS_TEST_P(uct_amo_fand_for_test, fand32) {
+    check_atomics(UCS_BIT(UCT_ATOMIC_OP_AND), FOP32);
```

Comment:
minor: maybe check_atomics can be moved inside test_op/test_fop to reduce code

Reply:
bit refactored test_op/fop for and/or ops (add/xor will update after migration to new API) - now atomics are evaluated in test_op/fop call